docs/blog/2020-03-11-netlify-functions-and-gatsby-cloud/index.md

```diff
@@ -0,0 +1,164 @@
+
+Case in point: when I was rebuilding my [personal blog](https://joshwcomeau.com/), I wanted to track and display the number of likes each article gets, as well as the number of hits (for a retro-style hit counter).
+
+For these kinds of situations, _serverless functions_ are perfect. They let us write small bits of Node.js code without worrying about where that code will run.
+
+My personal blog is built and deployed with [Gatsby Cloud](https://www.gatsbyjs.com/), a CI service for Gatsby sites, and it's hosted by [Netlify](https://www.netlify.com/). I'm a very happy Netlify customer, and [Netlify Functions](https://www.netlify.com/products/functions/) seemed like the perfect service for my needs!
+
+Getting Gatsby Cloud and Netlify Functions to cooperate took a bit of tinkering, but happily it can be done! The solution I discovered feels robust and reliable, and my blog has been powered by these two services for several weeks now, without any issues.
+
+Today we'll look at how to use Netlify Functions for your Gatsby Cloud site.
+
+# What are Netlify Functions?
+
+Netlify functions let you run event-driven server-side code, without worrying about running and maintaining a server. It's a service that sits in front of AWS Lambda, and brushes away some of the thorns of working directly with Amazon Web Services.
+
+It allows you to write code like this, used by my blog to track hits:
+
+```js
+const faunadb = require("faunadb")
+
+exports.handler = async (event, context, callback) => {
+  // Connect to the database
+  const q = faunadb.query
+  const client = new faunadb.Client({
+    secret: process.env.FAUNADB_SECRET,
+  })
+
+  // Get the provided slug, and use it to look up
+  // the corresponding document.
+  const { slug } = JSON.parse(event.body)
+  const document = await client.query(
+    q.Get(q.Match(q.Index("hits_by_slug"), slug))
+  )
+
+  // Increment the number of hits by 1
+  await client.query(
+    q.Update(document.ref, {
+      data: { hits: document.data.hits + 1 },
+    })
+  )
+
+  return {
+    statusCode: 200,
+    body: JSON.stringify({
+      success: true,
+    }),
+  }
+}
+```
+
+When using Netlify for CI _and_ deployments, you can pop this code in a `/functions` directory and the functions will get built and shipped whenever you push to Github. No manual steps needed 💯
+
+Learn more about Netlify Functions in [their documentation](https://docs.netlify.com/functions/overview/).
+
+# What is Gatsby Cloud?
+
+Gatsby Cloud is a service provided by Gatsby Inc. to manage deployments of Gatsby apps.
+
+Critically, Gatsby Cloud is _not_ an alternative to Netlify—we still use CDN providers like Netlify or Amazon S3 to host your projects. Rather, Gatsby Cloud is a CI service that builds your site and distributes it for you.
+
+<img
+  src={netlifyFunctions}
+  caption="A screenshot of the Gatsby Cloud project dashboard. Need to work on my site's perf!"
```

Comment:
How does this image `caption` attribute get applied? That seems nonstandard–is it a custom component?

Reply:
Ack good catch! I copy/pasted the markdown from my personal blog, which is indeed a custom component.

Will remove!